src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -942,13 +942,14 @@ static UCS_CLASS_CLEANUP_FUNC(uct_dc_mlx5_ep_t)
 
     ucs_assert_always(self->flags & UCT_DC_MLX5_EP_FLAG_VALID);
 
-    if (iface->flags & UCT_DC_MLX5_IFACE_FLAG_KEEPALIVE) {
+    if (self->flags & UCT_DC_MLX5_EP_FLAG_KEEPALIVE_POSTED) {
```

Comment:
maybe add helper function:
uct_dc_mlx5_ep_keepalive_cleanup
for lines 945-955

Reply:
added helper function